Formula/t/typedb.rb

```diff
@@ -12,6 +10,27 @@ class Typedb < Formula
 
   depends_on "openjdk"
 
+  on_arm do
+    on_macos do
+      url "https://github.com/typedb/typedb/releases/download/2.29.1/typedb-all-mac-arm64-2.29.1.zip"
+      sha256 "1270565acd7d5c61475831dac408f2069533ecf4ffee416ae474962ce1a71603"
+    end
+    on_linux do
+      url "https://github.com/typedb/typedb/releases/download/2.29.1/typedb-all-linux-arm64-2.29.1.tar.gz"
+      sha256 "4846e0496c9d90542fe677bd44ec78fe31a056a07770b5b53703ea0c781e99d6"
+    end
+  end
+  on_intel do
+    on_macos do
+      url "https://github.com/typedb/typedb/releases/download/2.29.1/typedb-all-mac-x86_64-2.29.1.zip"
+      sha256 "82ad962d3248d0a883d129a01b7593960031758283270128601a948be637854a"
+    end
+    on_linux do
+      url "https://github.com/typedb/typedb/releases/download/2.29.1/typedb-all-linux-x86_64-2.29.1.tar.gz"
+      sha256 "4b358ee8beb76f856ca21c61979828f9e082f00591942bf770a1cb2aa53fe4bf"
+    end
+  end
```

Comment:
Looks suspicious TBH. Doesn't seem compatible with https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae

Reply:
Diffing the extract files and running `pkgdiff` on the differing jar files, there are currently two that contain OS/arch-specific dynamic libraries:

* console/lib/com-vaticle-typedb-typedb-driver-jni-(linux|macosx)-(arm64|x86_64)-2.29.0.jar
* server/lib/com-google-ortools-ortools-(darwin|linux) -(aarch64|x86-64)-9-6-2534.jar

The 2.23.0 release provides both the ARM and x86 ortools files and it doesn't contain the aforementioned `driver-jni` file.

There are other jar files in server/lib (e.g., com-vaticle-typedb-typedb-core-common-2.29.1.jar) that differ only in `META-INF/MANIFEST.MF` (e.g., `Target-Label: //common:common-mac-arm64` vs. `Target-Label: //common:common-linux-x86_64`). The 2.23.0 release omits the arch from the `Target-Label` (e.g., `common-mac`, `common-linux`).

Reply:
Could try using Bazel/Bazelisk, but there is a chance it will end up downloading the pre-built JARs.

3rd party JARs are most likely downloaded. We don't have a way to deal with these (same situation when you build from Maven or Gradle).

There should be some upstream workflow that is handling 1st party JARs (the com.vaticle ones).